[Embeddable] SLO Alerts embeddable to serialized state only

nickpeihl opened the pull request:
_This PR does not need to be reviewed by external teams. This PR merges into a feature branch that Kibana presentation team is working on to convert the embeddable framework to only expose serialized state. Your team will be pinged for review once the work is complete and the final PR opens that merges the feature branch into main._

## Summary
Converts the SLO alerts embeddable to serialized state only.

## Testing this PR

Create an SLO using the "How to Test" section in the description of [this PR](https://github.com/elastic/kibana/pull/179147).

nickpeihl committed:
Convert SLO Alerts embeddable to serialized state only

nickpeihl committed:
Remove slo state manager from api

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293452)
* Commit: 88568929417f6acc2b8f43ff42c7f50ac302d282

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/293452#01963b40-c7ce-4c68-9225-02702d1c4cd1)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293452","number":293452,"commit":"88568929417f6acc2b8f43ff42c7f50ac302d282"}],"number":293452}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293462)
* Commit: 1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/293462#01963b60-2a76-49af-8009-2977109da4b2)

### History
* :broken_heart: [Build #293455](https://buildkite.com/elastic/kibana-pull-request/builds/293455) failed 55d8a635922cac4694b1f5a7658b12f22d775630
* :broken_heart: [Build #293452](https://buildkite.com/elastic/kibana-pull-request/builds/293452) failed 88568929417f6acc2b8f43ff42c7f50ac302d282

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293462","number":293462,"commit":"1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293455","number":293455,"commit":"55d8a635922cac4694b1f5a7658b12f22d775630"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293452","number":293452,"commit":"88568929417f6acc2b8f43ff42c7f50ac302d282"}],"number":293462}
buildkite-pr-comment-->

nreese commented on x-pack/solutions/observability/plugins/slo/public/embeddable/slo/alerts/slo_alerts_embeddable_factory.tsx:
```diff
@@ -72,52 +68,68 @@ export function getAlertsEmbeddableFactory({
+        ...titleManager.api,
+        ...unsavedChangesApi,
+        defaultTitle$,
+        hideTitle$: titleManager.api.hidePanelTitles$,
```
What is `hideTitle$` and `setHideTitle` why are they added to the api? Title apis should be provided by `...titleManager.api,`

nreese replied:
You should not longer need these. https://github.com/elastic/kibana/pull/218369 has resolved the issue and now provides `hideTitle$` and `setHideTitle` in the `titleManager.api`

nickpeihl replied:
Awesome! Thanks for updating the titleManager.

dej611 commented on x-pack/platform/plugins/shared/lens/public/react_embeddable/lens_embeddable.tsx:
Why all these changes to the Lens embeddable too? Are they related to the SLO Alerts ones?

nreese replied:
We are working on a feature branch to convert all embeddables to serialized state only. Lens has been converted but there are some eslinting errors in the branch. This PR is resolving those. Please ignore review requests on the branch. Once we get all embeddables converted and can get green CI, then we will open a PR for review. This will be in about 2 or 3 weeks time

nickpeihl committed:
Remove hideTitle$ and setHideTitle from api. Already provided by  titleManager.api

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293735)
* Commit: 7d730d87188ced56812a5066c2c125f99d2e373c

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/293735#01963eb2-c129-4f8b-98cb-ccf2eea044c5)

### History
* :broken_heart: [Build #293462](https://buildkite.com/elastic/kibana-pull-request/builds/293462) failed 1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65
* :broken_heart: [Build #293455](https://buildkite.com/elastic/kibana-pull-request/builds/293455) failed 55d8a635922cac4694b1f5a7658b12f22d775630
* :broken_heart: [Build #293452](https://buildkite.com/elastic/kibana-pull-request/builds/293452) failed 88568929417f6acc2b8f43ff42c7f50ac302d282

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293735","number":293735,"commit":"7d730d87188ced56812a5066c2c125f99d2e373c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293462","number":293462,"commit":"1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293455","number":293455,"commit":"55d8a635922cac4694b1f5a7658b12f22d775630"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293452","number":293452,"commit":"88568929417f6acc2b8f43ff42c7f50ac302d282"}],"number":293735}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

nickpeihl committed:
[Serialized State Only] Field list embeddable (#218174)

_This PR does not need to be reviewed by external teams. This PR merges
into a feature branch that Kibana presentation team is working on to
convert the embeddable framework to only expose serialized state. Your
team will be pinged for review once the work is complete and the final
PR opens that merges the feature branch into main._

Updates the Field list example embeddable to serialized state only.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nickpeihl committed:
Merge branch 'embeddable-serialized-state' into slo-alerts-sso2

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293791)
* Commit: 9d8220bf3e6b0cc30badd1365f9e3bac8dd703c0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/293791#01963eef-594c-42a6-b88b-f00927a1f45b)

### History
* :broken_heart: [Build #293769](https://buildkite.com/elastic/kibana-pull-request/builds/293769) failed 4a0b96383cfa55489b2d09f87a1f339579d457e3
* :broken_heart: [Build #293755](https://buildkite.com/elastic/kibana-pull-request/builds/293755) failed b58e7ddab59903d296ebd207868ba91c996c271e
* :broken_heart: [Build #293735](https://buildkite.com/elastic/kibana-pull-request/builds/293735) failed 7d730d87188ced56812a5066c2c125f99d2e373c
* :broken_heart: [Build #293462](https://buildkite.com/elastic/kibana-pull-request/builds/293462) failed 1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65
* :broken_heart: [Build #293455](https://buildkite.com/elastic/kibana-pull-request/builds/293455) failed 55d8a635922cac4694b1f5a7658b12f22d775630

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293791","number":293791,"commit":"9d8220bf3e6b0cc30badd1365f9e3bac8dd703c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293769","number":293769,"commit":"4a0b96383cfa55489b2d09f87a1f339579d457e3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293755","number":293755,"commit":"b58e7ddab59903d296ebd207868ba91c996c271e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293735","number":293735,"commit":"7d730d87188ced56812a5066c2c125f99d2e373c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293462","number":293462,"commit":"1ee4dcf68fa80d2d2d5ab43f9ae727924f9d3d65"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293455","number":293455,"commit":"55d8a635922cac4694b1f5a7658b12f22d775630"}],"number":293791}
buildkite-pr-comment-->
